### ISSUE TITLE
Changes that are made to an entry draft while autosave is in progress are sometimes not saved

### ISSUE THREAD

**Issue by linusschwab** [craftcms/cms#5305]
### Description

There is a bug in the entry autosave feature: if you make changes to an entry while autosave is in progress, the last changes are sometimes not saved. If you are still typing after the save process is completed the entry is saved again after a short delay. However, if you finished typing during this time, the changes are not saved again.

Craft CMS detects that there are unsaved changes after this if you try to switch the page and warns you. Therefore, the solution could be to just add this check after the save process is completed and save again if necessary.

### Steps to reproduce

1. Create an entry with a matrix field and a couple of blocks (around 4-5)
2. Change some content, for example in a text or redactor field
3. While the saving icon is spinning, add another character

### Additional info

- Craft version: 3.3.17
- PHP version: 7.2.16
- Database driver & version: MySQL 5.7.25
- Plugins & versions: Element API (2.6.0), Redactor (2.4.0), Super Table (2.3.1)

**Comment by brandonkelly** [comment]
I’m not able to reproduce this, but suspect I may have found the culprit anyway. Can you please test commit c37c7df2af96ef63b7c9b1ec77d98d002f823a72 and see if it resolves the issue for you?

To get the fix early, change your `craftcms/cms` requirement in composer.json to:

```json
"require": {
  "craftcms/cms": "dev-develop#c37c7df2af96ef63b7c9b1ec77d98d002f823a72 as 3.3.17",
  "...": "..."
}
```

Then run `composer update`.

**Comment by linusschwab** [comment]
Thank you for the fast response. Unfortunately the fix does not solve the problem.

Did you try to reproduce the issue with a more complex entry? Maybe this makes it easier to reproduce if the saving takes a bit longer. However, I'm also able to reproduce the problem with a simple entry type with only two text fields and one image field.

**Comment by brandonkelly** [comment]
Good idea. I set an Xdebug breakpoint in the `entry-revisions/save-draft` controller action, so that I could be sure that a second change was made while the first change was still saving, and with that I was able to reproduce the issue.

To get the fix early, change your `craftcms/cms` requirement in composer.json to:

```json
"require": {
  "craftcms/cms": "dev-develop#8c69daf66bad5c4b05e94a6744ba44f49f27cdbf as 3.3.17",
  "...": "..."
}
```

Then run `composer update`.
